raster/r.report/testsuite/test_r_report.py

```diff
@@ -58,6 +64,613 @@ def test_output(self):
+                                {"unit": "square kilometers", "value": 0.617},
+                                {"unit": "acres", "value": 152.4640203652408},
+                                {"unit": "hectares", "value": 61.7},
+                                {"unit": "cell counts", "value": 6170},
+                                {"unit": "% cover", "value": 2.9523084947054627},
+                            ],
+                        },
+                        {
+                            "category": 2,
+                            "label": "from to",
+                            "range": {
+                                "from": 105.9543285369873,
+                                "to": 156.32986450195312,
+                            },
+                            "units": [
+                                {"unit": "square miles", "value": 7.8308853693420355},
+                                {"unit": "square meters", "value": 20281900},
+                                {"unit": "square kilometers", "value": 20.2819},
+                                {"unit": "acres", "value": 5011.766636378894},
+                                {"unit": "hectares", "value": 2028.19},
+                                {"unit": "cell counts", "value": 202819},
+                                {"unit": "% cover", "value": 97.04769150529454},
+                            ],
+                        },
+                    ],
+                },
+            ],
+            "totals": [
+                {"unit": "square miles", "value": 78.18568710484531},
+                {"unit": "square meters", "value": 202500000},
+                {"unit": "square kilometers", "value": 202.5},
+                {"unit": "acres", "value": 50038.839747100916},
+                {"unit": "hectares", "value": 20250},
+                {"unit": "cell counts", "value": 2025000},
+                {"unit": "% cover", "value": 100},
+            ],
+        }
+        module = SimpleModule(
+            "r.report",
+            map="towns,elevation",
+            units=[
+                "miles",
+                "meters",
+                "kilometers",
+                "acres",
+                "hectares",
+                "cells",
+                "percent",
+            ],
+            nsteps=2,
+            format="json",
+        )
+        self.runModule(module)
+        data = json.loads(module.outputs.stdout)
+
+        # created field represents the time of running the command. therefore, its exact value
+        # cannot be tested. we only check that it is present and in the ISO8601 datetime format
+        self.assertIn("created", data)
+        try:
+            # on python 3.11 and below, datetime.fromisoformat doesn't support zone info with offset
+            datetime.strptime(data["created"], "%Y-%m-%dT%H:%M:%S%z")
+        except ValueError:
+            self.fail("created field is not in isoformat: %s" % (data["created"],))
```

Comment:
We have a script used when building that handles that case, we changed it this spring exactly for that reason.
Does that mean that the test only really checks when it's on Python 3.12+?

See https://github.com/OSGeo/grass/pull/3721

Reply:
Ah, no this one checks it always. The comment was to tell why I didn't use datetime.fromisoformat function from Python's standard library and instead used datetime.strptime.